lib/api/objectPutPart.js

```diff
@@ -336,19 +337,55 @@ function objectPutPart(authInfo, request, streamingV4Params, log,
                         });
                         return next(err, destinationBucket);
                     }
-                    return next(null, oldLocations, objectLocationConstraint,
+                    return next(null, partLocations, oldLocations, objectLocationConstraint,
+                        destinationBucket, hexDigest, prevObjectSize, splitter);
+                });
+        },
+        (partLocations, oldLocations, objectLocationConstraint, destinationBucket,
```

Comment:
It might make sense to move this into a helper so that we can us it in PutCopyPart as well rather than duplicating the logic. The wrapper code that would be needed may outweigh the benefits though.

Reply:
Yeah, I can do a `isCompleteMPUInProgress` in services for example. I would likely still have to duplicate the "not deleting locations" log since creating a helper to encapsulate the deletion of the parts would be too convoluted in the current logic or would require a larger refactor to encapsulate more logic from `objectPutPart`/`objectPutCopyPart`.

Reply:
done